odbc/src/api.rs

```diff
@@ -243,46 +267,64 @@ pub extern "C" fn SQLColumnsW(
 #[no_mangle]
 pub extern "C" fn SQLCompleteAsync(
     _handle_type: HandleType,
-    _handle: Handle,
+    handle: Handle,
     _async_ret_code_ptr: *mut RetCode,
 ) -> SqlReturn {
-    unimplemented!()
+    unsafe {
+        (*(handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLCompleteAsync".to_string()));
+        SqlReturn::ERROR
+    }
 }
 
 #[no_mangle]
 pub extern "C" fn SQLConnect(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _server_name: *const Char,
     _name_length_1: SmallInt,
     _user_name: *const Char,
     _name_length_2: SmallInt,
     _authentication: *const Char,
     _name_length_3: SmallInt,
 ) -> SqlReturn {
-    unimplemented!()
+    unsafe {
+        (*(connection_handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLConnect".to_string()));
+        SqlReturn::ERROR
+    }
 }
 
 #[no_mangle]
 pub extern "C" fn SQLConnectW(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _server_name: *const WChar,
     _name_length_1: SmallInt,
     _user_name: *const WChar,
     _name_length_2: SmallInt,
     _authentication: *const WChar,
     _name_length_3: SmallInt,
 ) -> SqlReturn {
-    unimplemented!()
+    unsafe {
+        (*(connection_handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLConnectW".to_string()));
+        SqlReturn::ERROR
+    }
 }
 
 #[no_mangle]
-pub extern "C" fn SQLCopyDesc(_source_desc_handle: HDesc, _target_desc_handle: HDesc) -> SqlReturn {
-    unimplemented!()
+pub extern "C" fn SQLCopyDesc(source_desc_handle: HDesc, target_desc_handle: HDesc) -> SqlReturn {
+    unsafe {
+        (*(source_desc_handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLCopyDesc".to_string()));
```

Comment:
Oh good catch! Not sure if we need a test though, since the `_source_desc_handle` parameter is now prefixed with an underscore, which makes it clear that only the target handle gets updated?

Update: I'm actually going to leave this function implementation as "unimplemented!()" now that I've filed [SQL-661](https://jira.mongodb.org/browse/SQL-661)